File: GeodeticUTMConverterTests/UTMConverterTests.m
//
//  UTMConverterTests.m
//  UTMConverterTests
//
//  Created by Cameron Lowell Palmer & Mariia Ruchko on 19.06.12.
//  Copyright (c) 2012 Cameron Lowell Palmer & Mariia Ruchko. All rights reserved.
//

#import "UTMConverterTests.h"

#import "GeodeticUTMConverter.h"



@implementation UTMConverterTests

- (void)setUp {
    [super setUp];
}

- (void)tearDown {
    [super tearDown];
}

- (void)testUTMToLatLong {
    UTMCoordinates coordinates;
    coordinates.easting = 598430;
    coordinates.northing = 6643010;
    coordinates.gridZone = 32;
    coordinates.hemisphere = kUTMHemisphereNorthern;
    
    GeodeticUTMConverter *utmConverter = [[GeodeticUTMConverter alloc] init];
    CLLocationCoordinate2D latLong = [utmConverter UTMCoordinatesToLatitudeAndLongitude:coordinates];
    [utmConverter release];
    
    NSLog(@"Latitude: %f, Longitude: %f", latLong.latitude, latLong.longitude);
    STAssertTrue(latLong.latitude == 59.912814611065265, @"Should be equal.");
    STAssertTrue(latLong.longitude == 10.760192985178369, @"Should be equal.");
}

- (void)testLatLongToUTM {
    CLLocationCoordinate2D coordingates;
    coordingates.latitude = 59.912814611065265;
    coordingates.longitude = 10.760192985178369;

    GeodeticUTMConverter *utmConverter = [[GeodeticUTMConverter alloc] init];
    UTMCoordinates utmCoordinates = [utmConverter latitudeAndLongitudeToUTMCoordinates:coordingates];
    [utmConverter release];

    NSString *hemisphere;
    if (utmCoordinates.hemisphere == kUTMHemisphereNorthern) {
        hemisphere = @"Northern";
    } else {
        hemisphere = @"Southern";
    }
    
    NSLog(@"Northing: %f, Easting: %f, GridZone: %d, Hemisphere: %@", utmCoordinates.northing, utmCoordinates.easting, utmCoordinates.gridZone, hemisphere);
    STAssertTrue((int)utmCoordinates.northing == 6643010, @"Should be equal.");
    STAssertTrue((int)utmCoordinates.easting == 598430, @"Should be equal.");
    STAssertTrue(utmCoordinates.gridZone == 32, @"");
    STAssertTrue(utmCoordinates.hemisphere == kUTMHemisphereNorthern, @"");
}

@end

File: GeodeticUTMConverterTests/UTMConverterTests.swift
//
//  UTMConverterTests.swift
//  UTMConverterTests
//
//  Created by Cameron Palmer on 22.10.2015.
//  Copyright © 2015 Bird and Bear Productions. All rights reserved.
//

import XCTest
import CoreLocation
@testable import UTMConverter

class UTMConverterTests: XCTestCase {

    override func setUp() {
        super.setUp()
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }

    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }

    func testUTMToLocation() {
        let coordinate = UTMCoordinate(easting: 598430, northing: 6643010, gridZone: 32, latitudeBand: UTMLatitudeBand.A, hemisphere: UTMHemisphere.Northern)

        let converter = UTMConverter()
        let location = converter.UTMToLocation(coordinate)

        NSLog("Latitude: %f, Longitude: %f", location.latitude, location.longitude);
        XCTAssertEqualWithAccuracy(location.latitude, 59.912814611065265, accuracy: 0.001, "Should be equal.");
        XCTAssertEqualWithAccuracy(location.longitude, 10.760192985178369, accuracy: 0.001, "Should be equal.");
    }

    func testLocationToUTM() {
        let location = CLLocationCoordinate2DMake(59.912814611065265, 10.760192985178369)

        let converter = UTMConverter()
        let UTM = converter.locationToUTM(location)

        var hemisphere: String;
        if (UTM.hemisphere == UTMHemisphere.Northern) {
            hemisphere = "Northern";
        } else {
            hemisphere = "Southern";
        }

        NSLog("Northing: %f, Easting: %f, GridZone: %d, Hemisphere: %@", UTM.northing, UTM.easting, UTM.gridZone, hemisphere);
        XCTAssertEqualWithAccuracy(UTM.northing, 6643010, accuracy: 0.1, "Should be equal.");
        XCTAssertEqualWithAccuracy(UTM.easting, 598430, accuracy: 0.1, "Should be equal.");
        XCTAssertEqual(UTM.gridZone, 32, "");
        XCTAssertEqual(UTM.hemisphere, UTMHemisphere.Northern, "");
    }
}
